Add properties object to Grants & update hashing algorithm (breaking)

ronaldkoster opened the pull request:
1. Each Grant can contain an optional `properties` field that can be used to provide extra information. This makes it easier to extend FSC with additional functionality. Eg. to ensure correct authentication/authorization.
2. Update hashing algorithm to use JSC. This is a breaking change.

Voles commented on specifications.md:
```diff
@@ -197,6 +201,30 @@ Signature requirements:
+
+#### Security Considerations
+
+- The size of the `properties` object should be limited to prevent excessive data transfer and storage. It is recommended to implement a size limit of 1MB for the serialized properties object.
```
```suggestion
- It is **RECOMMENDED** to implement a size limit of 1MB for the serialized properties object. This limit should prevent excessive data transfer and storage.
```

Voles commented on specifications.md:
```diff
@@ -530,6 +563,7 @@ Before issuing an access token the Manager **MUST** validate that:
 The `cnf.x5t#S256` claim **MUST** contain the certificate thumbprint of the X.509 certificate provided by the client requesting the token according to [section 3.1] of [[RFC8705]].
 The `act` claim **MUST** be set when an access token is generated for a Peer who is connecting to the Service on behalf of another Peer. I.e. the authorization to connect has been granted using a DelegatedServiceConnectionGrant.
 The `pdi` claim **MUST** be set when an access token is generated for a Service which is being offered on behalf of another Peer. 
+The `prp` claim **MUST** be set when the Grant contains a `properties` object in the field `grant.data.properties`. 
```
```suggestion
The `prp` claim **MUST** be set when the Grant contains a the `properties` object for the field `grant.data.properties`. 
```

ronaldkoster replied:
I Used a different sentence structure.

RonaldKoster86 committed:
add properties object to grants

Each Grant can contain an optional properties object that can be used to provide extra information.
This makes it easier to extend FSC with additional functionality.

Voles commented on specifications.md:
```diff
@@ -301,44 +301,25 @@ The Validation **MUST** be done every time a Peer receives a signature.
+1. Add the fields `iv` and `group_id` to the `data` object of the Grant. 
+1. Set `contract.content.group_id` as value for `data.group_id`.
+1. Set `contract.content.iv` as value for `data.iv`.
+1. Convert the Grant to Canonical JSON data as described in [[RFC8785]]
```
Let's be explicit here if `{ Data: { ... }}` should be hashed, or, only the contents of `Data`.

RonaldKoster86 committed:
update hash creation instructions

Use RFC8785 JSON Canonicalization Scheme to create consistent JSON objects.
Because the objects are consistent they create consitent hashes.
By using the RFC FSC core no longer needs to define how the JSON objects should be handled to create consistent hashes.

Voles committed:
Add details about the JSON format to the protocol section

Voles committed:
Remove dots at end of the line for consistency

Voles committed:
Remove duplicate details about the usage of JCS

Voles committed:
Formatting

Voles committed:
Add clarification about the allowed properties

RonaldKoster86 committed:
update grant hash section

make it more clear that the Canonical JSON should represented as a string

TheBonheurs committed:
Resolve conflicts with develop

sanderke committed:
Fix ref
